binary_search_tree/tree.py

```diff
@@ -1,59 +1,119 @@
 class TreeNode:
-    def __init__(self, key, val = None):
+    def __init__(self, key, val=None):
         if val == None:
             val = key
 
         self.key = key
         self.value = val
         self.left = None
         self.right = None
-        
 
 
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
+    def add_helper(self, current_node, key, value):
+        if current_node is None:
+            return TreeNode(key, value)
 
-    # Time Complexity: 
-    # Space Complexity: 
+        if key < current_node.key:
+            current_node.left = self.add_helper(current_node.left, key, value)
+        else:
+            current_node.right = self.add_helper(
+                current_node.right, key, value
+            )
+
+        return current_node
+
+    # Time Complexity: O(log n)
+    # Space Complexity: O(n)
+    def add(self, key, value=None):
```

Comment:
✨ Time complexity and space complexity will both be O(log n) if balanced and O(n) if unbalanced